### ISSUE TITLE
Missing rights on ApplicationSet resource for application controller service account

### ISSUE THREAD

**Issue by OpenGuidou** [argoproj/argo-helm#2974]
### Describe the bug

Hello,

When performing an argocd installation using the chart, namespace-scoped (createClusterRoles: false), we can't sync ApplicationSet in the cluster as the role defined for the application controller is missing this resource.
An error is displayed as the application controller can't create applicationsets

### Related helm chart

argo-cd

### Helm chart version

7.6.8

### To Reproduce

1. Install the chart argo-cd with value createClusterRoles: false
2. Create an app referencing a repo containing an applicationset
3. See the error as in the screenshot below

### Expected behavior

The applicationset should be correctly created.

### Screenshots

![image](https://github.com/user-attachments/assets/41d714ac-a0a2-494f-92ae-c6b3813de1fe)

**Comment by OpenGuidou** [comment]
It seems that even by adding the applicationset resource in the application-controller role, other resources rights are missing:

![image](https://github.com/user-attachments/assets/ced6470f-76b4-40e3-bdb5-ddb96fbbd8d1)

**Comment by crenshaw-dev** [comment]
I'm not sure that this is actually unexpected behavior. My understanding of namespaced mode is that Argo CD comes with roles for each component to manage their own resources and _do nothing else_. There's no GitOps management out of the box. After installing the namespaced-mode Argo CD, you're expected to add cluster configurations for anywhere you want to deploy to.

I'm inferring all this from looking at the manifests, I could be missing something.

**Comment by OpenGuidou** [comment]
One of the usual use cases it to have ArgoCD deployed on a control plane cluster, managing applicative remote clusters.
In my case, we have several namespaced instances, each managing their set of clusters.

In such case, we expect to deploy in-cluster only Applications and ApplicationSets. Other ressources are deployed on the target remote clusters.
I would find it difficult to explain that we can use applications and not applicationsets.

I think this use case is common enough to not force all users to redefine their roles on top to allow it.

**Comment by crenshaw-dev** [comment]
Fair point, especially when it's possible to create (certain) AppSets via the API. 

If the goal is to allow people to fully manage Argo CD's own config out of the box, we'd want to add write access for these as well:

```
grep -E "^kind: " manifests/namespace-install.yaml | uniq | sort | awk '{print $2}'
ConfigMap
Deployment
NetworkPolicy
Role
RoleBinding
Secret
Service
ServiceAccount
StatefulSet
```

If the goal is feature parity between API-managed resources and GitOps managed resources, I think the list is this:

```
Applications
ApplicationSets
AppProjects
Secrets
```

I've added a topic to the next [contributors meeting](https://docs.google.com/document/d/1xkoFkVviB70YBzSEa4bDnu-rUZ1sIFtwKKG1Uw8XsY8/edit?tab=t.0). We'll pick a strategy and then make the changes. Probably need some docs around this.

**Comment by OpenGuidou** [comment]
Should we list secrets in the gitops option ?
Are there still people pushing secrets in git ?  😁
Could we advise ExternalSecret or no secrets at all if we go for a solution without external dependency?

**Comment by OpenGuidou** [comment]
After a few checks, we should probably also add the rights watch + list on configmaps and secrets.

Application log controller without:
```
Failed to watch *v1.Secret: failed to list *v1.Secret: secrets is forbidden: User "system:serviceaccount:argocd:argocd-application-controller" cannot list resource "secrets" in API group "" in the namespace "argocd"
Failed to list *v1.Secret: secrets is forbidden: User "system:serviceaccount:argocd:argocd-application-controller" cannot list resource "secrets" in API group "" in the namespace "argocd"

```

**Comment by crenshaw-dev** [comment]
Agreed on Secrets.

Contributors call says lets add Applications access.

We probably need some upstream docs on the _intent_ of namespaced mode. i.e. you can GitOps-managed Argo CRDs in the `argocd` namespace out of the box, but for anything else you're going to have to configure your own access.

**Comment by OpenGuidou** [comment]
Up

**Comment by OpenGuidou** [comment]
Merged upstream, pending release
